opencti-platform/opencti-graphql/src/modules/publicDashboard/publicDashboard-utils.ts

```diff
@@ -36,16 +40,20 @@ export const getWidgetArguments = async (
     throw UnsupportedError('User not found');
   }
 
-  // To replace User markings by publicDashboard allowed_markings
+  // Determine the marking definitions allowed.
   const allMarkings = await getEntitiesListFromCache<StoreMarkingDefinition>(context, SYSTEM_USER, ENTITY_TYPE_MARKING_DEFINITION);
+  const maxMarkings = await getMaxMarkings(context, SYSTEM_USER);
+  const availableMaxMarkings = computeAvailableMarkings(maxMarkings, allMarkings).map((m) => m.id);
+  const availableDashboardMarkings = computeAvailableMarkings(allowed_markings, allMarkings);
+  const availableMarkings = availableDashboardMarkings.filter((m) => availableMaxMarkings.includes(m.id));
```

Comment:
I think that all this block starting line 44 could be extracted in a dedicated method. I find it a bit difficult to understand. And part of it is duplicated in `publicDashboard-domain.ts`.

Reply:
Can you tell me if you find it easier to read now?